addons/im_livechat/report/im_livechat_report_operator.py

```diff
@@ -35,12 +35,11 @@ def init(self):
                     C.create_date as start_date,
                     EXTRACT('epoch' FROM (max((SELECT (max(M.create_date)) FROM mail_message M JOIN mail_message_mail_channel_rel R ON (R.mail_message_id = M.id) WHERE R.mail_channel_id = C.id))-C.create_date)) as duration,
                     EXTRACT('epoch' from ((SELECT min(M.create_date) FROM mail_message M, mail_message_mail_channel_rel R WHERE M.author_id=P.id AND R.mail_channel_id = C.id AND R.mail_message_id = M.id)-(SELECT min(M.create_date) FROM mail_message M, mail_message_mail_channel_rel R WHERE M.author_id IS NULL AND R.mail_channel_id = C.id AND R.mail_message_id = M.id))) as time_to_answer
-                FROM im_livechat_channel_im_user O
-                    JOIN res_users U ON (O.user_id = U.id)
-                    JOIN res_partner P ON (U.partner_id = P.id)
-                    JOIN im_livechat_channel L ON (L.id = O.channel_id)
-                    JOIN mail_channel C ON (C.livechat_channel_id = L.id)
-                    JOIN mail_channel_partner MCP ON (MCP.partner_id = P.id AND MCP.channel_id = C.id)
+                FROM mail_channel C
+                    JOIN mail_channel_partner S ON (S.channel_id = C.id and S.partner_id = C.livechat_operator_id)
```

Comment:
I think many things, but this one can be cool:
We simply need `livechat_operator_id` to determine the operator, we can remove res_partner, res_users and mail_channel_partner from the JOIN clause. We can make `LEFT JOIN` (since required).
Also, (give it a try), we can put mail_message in the JOIN clause (join it with the mail_message_mail_channel_rel table, and only keep the message with message_type=comment).
Then we can get the min/max create_date to compute `duration` and `time_to_answer`.